arangod/IResearch/IResearchPrimaryKeyFilter.cpp

```diff
@@ -58,9 +58,9 @@ bool PrimaryKeysFilter<Nested>::next() {
     auto const pkRef =
         irs::numeric_utils::numeric_traits<LocalDocumentId::BaseType>::raw_ref(
             pk);
-    auto doc = irs::doc_limits::eof();
+    auto doc = irs::doc_limits::invalid();
```

Comment:
eof also shouldn't be in segment
But this code is more correct and also condition with `0x0` instead of `0xffff`